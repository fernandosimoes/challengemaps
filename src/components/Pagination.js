import React from 'react';

const Pagination = ({changepage}) => {
    return (
        <div className="pagination">
            <button className="backpage" onClick={() => changepage('decrement')}>back</button>
            <button className="nextpage" onClick={() => changepage('increment')}>next</button>
        </div>
    )
}
export default Pagination;