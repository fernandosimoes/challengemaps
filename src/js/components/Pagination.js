import React from 'react';
import left from "../../assets/images/left.svg";
import right from "../../assets/images/right.svg";

const Pagination = ({changepage, currentPage, totalPages}) => {
    let pagesNumber = [];
    const nextpage = currentPage+1;
    const backpage = currentPage-1;
    for (let index = 0; index < totalPages; index++) {
        const selectedCLass = currentPage == index ? 'pagenumber selected' : 'pagenumber' ;

        pagesNumber.push(<button className={selectedCLass} key={index} onClick={() => changepage(index)}>
            {index + 1}
        </button>
        );
        // if (index == currentPage) {
        // }
        // if (index == backpage && backpage >= 0) {
        //     pagesNumber.push(<button className={selectedCLass} key={index} onClick={() => changepage(index)}>
        //         {index + 1}
        //     </button>
        //     );
        // }
        // if (index == nextpage && nextpage < totalPages) {
        //     pagesNumber.push(<button className={selectedCLass} key={index} onClick={() => changepage(index)}>
        //         {index + 1}
        //     </button>
        //     );
        // }
        
        
        
        // if (index == totalPages-1) {
        //     pagesNumber.push(<span>...</span> )
        //     pagesNumber.push(<button className={selectedCLass} key={index} onClick={() => changepage(index)}>
        //         {index + 1}
        //     </button>
        //     );
        // }
    }

    console.log('pagesNumber', pagesNumber)
    return (
        <div className="pagination">
            <button className="back" onClick={() => changepage(currentPage-1)}>
                <img src={left} />
            </button>
            {pagesNumber.map(page => page)}
            <button className="next" onClick={() => changepage(currentPage+1)}>
                <img src={right} />
            </button>
        </div>
    )
}

export default Pagination;