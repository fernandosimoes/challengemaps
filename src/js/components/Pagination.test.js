import React from 'react';
import Pagination from './Pagination';
import renderer from 'react-test-renderer';

test('Pagination', () => {

    let component = renderer.create(
        <Pagination changepage={()=>{}} currentPage={0} totalPages={10} />
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})
