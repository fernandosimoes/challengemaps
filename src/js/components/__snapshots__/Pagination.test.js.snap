// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination 1`] = `
<div
  className="pagination"
>
  <button
    className="back"
    onClick={[Function]}
  >
    <img
      src=""
    />
  </button>
  <button
    className="pagenumber selected"
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="pagenumber"
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="pagenumber"
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="pagenumber"
    onClick={[Function]}
  >
    4
  </button>
  <button
    className="pagenumber"
    onClick={[Function]}
  >
    5
  </button>
  <button
    className="pagenumber"
    onClick={[Function]}
  >
    6
  </button>
  <button
    className="pagenumber"
    onClick={[Function]}
  >
    7
  </button>
  <button
    className="pagenumber"
    onClick={[Function]}
  >
    8
  </button>
  <button
    className="pagenumber"
    onClick={[Function]}
  >
    9
  </button>
  <button
    className="pagenumber"
    onClick={[Function]}
  >
    10
  </button>
  <button
    className="next"
    onClick={[Function]}
  >
    <img
      src=""
    />
  </button>
</div>
`;
